### DISCUSSION TITLE
v0.8.5

### DISCUSSION THREAD

**Discussion by cgarciae** [google/flax#4031] [General]
## What's Changed
* v0.8.5 by @cgarciae in https://github.com/google/flax/pull/3941
* [nnx] improve vmap axis size detection by @cgarciae in https://github.com/google/flax/pull/3947
* Add direct penzai.treescope support for NNX objects. by @copybara-service in https://github.com/google/flax/pull/3948
* [nnx] fix nnx_basics dependencies by @cgarciae in https://github.com/google/flax/pull/3942
* Rename all the NNX tests to internal naming & build conventions. by @copybara-service in https://github.com/google/flax/pull/3952
* updated rng guide by @chiamp in https://github.com/google/flax/pull/3912
* upgraded haiku guide to include NNX by @chiamp in https://github.com/google/flax/pull/3923
* parameterized NNX transforms tests by @chiamp in https://github.com/google/flax/pull/3906
* Simplify extended dtypes rules part 1. Start by removing sharding specific rules from EDtypes. This is because we always want to replicate the trailing dims introduced by Edtypes. by @copybara-service in https://github.com/google/flax/pull/3957
* fix HEAD by @chiamp in https://github.com/google/flax/pull/3960
* Minor grammar fixes to NNX documentation. by @mcsmart76 in https://github.com/google/flax/pull/3953
* Make FlatState a Mapping instead of a dict by @NeilGirdhar in https://github.com/google/flax/pull/3928
* Adding Welford metric. by @copybara-service in https://github.com/google/flax/pull/3959
* Modify Welford metric to return mean value. by @copybara-service in https://github.com/google/flax/pull/3970
* [nnx] make State generic by @cgarciae in https://github.com/google/flax/pull/3964
* updated NNX nn docstrings by @chiamp in https://github.com/google/flax/pull/3972
* make flax work with upcoming JAX change to tree_map (being more careful about by @copybara-service in https://github.com/google/flax/pull/3976
* updated `nnx.module` docstrings by @chiamp in https://github.com/google/flax/pull/3966
* updated `nnx.Conv` and `nnx.ConvTranspose` by @chiamp in https://github.com/google/flax/pull/3974
* updated `nnx.graph` docstrings by @chiamp in https://github.com/google/flax/pull/3958
* * Adds `pmap` and `Pmap`. `static_broadcasted_argnums`, `donate_argnums`, and `global_arg_shapes` are not yet supported. by @copybara-service in https://github.com/google/flax/pull/3978
* Fixes for batch norm docs by @jkarwowski in https://github.com/google/flax/pull/3982
* fix deprecation warning by @chiamp in https://github.com/google/flax/pull/3981
* updated NNX `rnglib` docstring by @chiamp in https://github.com/google/flax/pull/3980
* updated `nnx.training` by @chiamp in https://github.com/google/flax/pull/3975
* updated `nnx.variables` docstrings by @chiamp in https://github.com/google/flax/pull/3986
* [nnx] vectorize vmap split counts by @cgarciae in https://github.com/google/flax/pull/3989
* added `wrt` option to `nnx.Optimizer` by @chiamp in https://github.com/google/flax/pull/3983
* Added `nnx.graph.iter_children` by @chiamp in https://github.com/google/flax/pull/3991
* [nnx] fix vmap by @copybara-service in https://github.com/google/flax/pull/3995
* Fix head pytest breakage by @IvyZX in https://github.com/google/flax/pull/4006
* Helper function for loading params from a linen module by @copybara-service in https://github.com/google/flax/pull/4012
* Port gemma/layers to NNX by @copybara-service in https://github.com/google/flax/pull/4013
* [nnx] fix grad by @cgarciae in https://github.com/google/flax/pull/4007
* [nnx] add PathContains Filter by @cgarciae in https://github.com/google/flax/pull/4011
* Support Python 3.9 by @copybara-service in https://github.com/google/flax/pull/4018
* Port gemma/modules to NNX by @copybara-service in https://github.com/google/flax/pull/4014
* Internal change to fix current head CI by @copybara-service in https://github.com/google/flax/pull/4017
* Unpin the Orbax pip version. by @copybara-service in https://github.com/google/flax/pull/4024
* Fix Gemma test to unbreak head by @IvyZX in https://github.com/google/flax/pull/4025
* Fix pickling of exceptions by @sanderland in https://github.com/google/flax/pull/4002
* Call user-defined variable transforms before determining axis size in nn.vmap. by @copybara-service in https://github.com/google/flax/pull/4026
* CI: add test run against oldest supported jax version by @jakevdp in https://github.com/google/flax/pull/3996
* Make `force_fp32_for_softmax` arg in `MultiHeadDotProductAttention` useful.  by @copybara-service in https://github.com/google/flax/pull/4029

## New Contributors
* @mcsmart76 made their first contribution in https://github.com/google/flax/pull/3953
* @jkarwowski made their first contribution in https://github.com/google/flax/pull/3982
* @sanderland made their first contribution in https://github.com/google/flax/pull/4002

**Full Changelog**: https://github.com/google/flax/compare/v0.8.4...v0.8.5

<hr /><em>This discussion was created from the release <a href='https://github.com/google/flax/releases/tag/v0.8.5'>v0.8.5</a>.</em>
